Assets/Scripts/Attributes/SingleLayer.cs

```diff
@@ -0,0 +1,6 @@
+﻿using UnityEngine;
+
+public class SingleLayer : PropertyAttribute
```

Comment:
Suggest restricting attribute usage

```suggestion
[AttributeUsage(AttributeTargets.Field | AttributeTargets.Property)]
public class SingleLayer : PropertyAttribute
```